MySQL migration script using Python (#27)
IBM-Cloud/vpc-migration-tools: db-migration/mysql/README.md

# MySQL database migration using Python

The MySQL database migration script allows you to migrate MySQL databases from one server to another. This is applicable to any platform:

* IBM classic infrastructure to IBM Cloud VPC
* On-premises to IBM Cloud VPC
* Other cloud service providers to IBM Cloud VPC

## Prerequisites

Review the following prerequisites before you begin your migration:

1. Set up an IBM Cloud Object Storage bucket.
2. Make sure that you have access to write to IBM Cloud Object Storage bucket.
3. Make sure that you have connection to the source and target server from your system.
4. Ensure that Python3 (version 3.0) and Pip3 are installed on your system.

> ## Cloning and running the database migration script

Reviewer: Do we need to call out where this script can be executed from (ie, separate device, serverless function)?

Author: No, its already mentioned as users system.

    ```

After running the database migration script `python3 db_migration.py`, you need to provide the details for the following parameters:
* Mount Object Storage bucket
* Source Server Details
* Source server database details
* Target server details
* Target server database details

## Mount Object Storage bucket

The IBM Cloud Object Storage bucket acts as centralized storage for the source and the target server. By using the `s3fs` utility, the bucket is mounted as a file system on both the source and target server. 
The source uses the bucket to store the database backup, and the target uses the bucket to retrieve the database backup, which minimizes the migration duration.

Complete the following steps to mount the IBM Cloud Object Storage bucket:

* COS Bucket name - Name of the bucket which has been already provisioned.

  ```
  Enter the Cloud Object Storage bucket name: `my-db-bucket`
  ```

* COS Endpoint - The COS endpoint helps to identify the location of the bucket.

  ```
  Enter the Cloud Object Storage endpoint.
  For Example: https://s3.dal.us.cloud-object-storage.appdomain.cloud
  COS Endpoint: https://s3.dal.us.cloud-object-storage.appdomain.cloud
  ```

> Hence the bucket is located on the US-South Dallas region. To know more about endpoints refer the below link:

Reviewer: ```suggestion
In this example, the bucket is located in the US-South Dallas region. To know more about endpoints refer the below link:
```
